src/mem3/src/mem3_rep.erl

```diff
@@ -454,21 +458,40 @@ changes_enumerator(#doc_info{id=DocId}, #acc{db=Db}=Acc) ->
     {ok, FDI} = couch_db:get_full_doc_info(Db, DocId),
     changes_enumerator(FDI, Acc);
 changes_enumerator(#full_doc_info{}=FDI, #acc{}=Acc0) ->
-    #acc{revcount = C, targets = Targets0, hashfun = HashFun} = Acc0,
+    #acc{
+        revcount = C,
+        targets = Targets0,
+        hashfun = HashFun,
+        incomplete_ranges = IncompleteRanges
+    } = Acc0,
     #doc_info{high_seq=Seq, revs=Revs} = couch_doc:to_doc_info(FDI),
     {Count, Targets} = case filter_doc(Acc0#acc.filter, FDI) of
-        keep -> {C + length(Revs), changes_append_fdi(FDI, Targets0, HashFun)};
-        discard -> {C, Targets0}
+        keep ->
+            NewTargets = changes_append_fdi(FDI, Targets0, HashFun,
+                IncompleteRanges),
+            {C + length(Revs), NewTargets};
+        discard ->
+            {C, Targets0}
     end,
     Acc1 = Acc0#acc{seq = Seq, revcount = Count, targets = Targets},
     Go = if Count < Acc1#acc.batch_size -> ok; true -> stop end,
     {Go, Acc1}.
 
 
-changes_append_fdi(#full_doc_info{id = Id} = FDI, Targets, HashFun) ->
+changes_append_fdi(#full_doc_info{id = Id} = FDI, Targets, HashFun,
+        IncompleteRanges) ->
     case mem3_reshard_job:pickfun(Id, maps:keys(Targets), HashFun) of
         not_in_range ->
-            Targets;
```

Comment:
Probably be a bit cleaner to add this as a guard  with `not_in_range when not IncompleteRanges ->` and similar.

Reply:
Good call. Used a guard for it and looks better